### ISSUE TITLE
MQTT input plugin does not accept events over 1KB of size

### ISSUE THREAD

**Issue by leoswaldo** [fluent/fluent-bit#9161]
## Bug Report

**Describe the bug**
If the payload of an event for the MQTT Input plugin is bigger that 1KB (1k characters) the messages are drop and not process at all, there is only the next message display the MQTT connection was closed

**To Reproduce**
1. Put the following defintions on fluent-bit.conf

```
[INPUT]
    Name   mqtt
    Tag    data
    Listen 0.0.0.0
    Port   1883

[OUTPUT]
    Name   stdout
    Match  *
```
2. Make the payload with over 1000 characters
 mosquitto_pub  -m '{"key1": 123, "key2": 1111111 to over a thousand of chaacters}' -t some/topic

**Expected behavior**
The over 1k characters events should be displayed in STDOUT, they are not.
If you run it for a small payload as the example in the MQTT it works but not for long payloads

**Your Environment**

* Version used: Present in 2.2 to 3.1
* Configuration: In steps to reproduce
* Environment name and version (e.g. Kubernetes? What version?): VM
* Server type and version: Centos9

**Comment by leonardo-albertovich** [comment]
I just opened PR #9163 which is scheduled to be included in version 3.1.5 and I'd appreciate it if you could give these changes a try. In the PR I've added a new setting named `buffer_size` which you can use to tune the maximum message length to your liking.

Thanks for taking the time to report this issue.

**Comment by edsiper** [comment]
closing since the solution has been merged on #9163
